command/jsonstate/state.go

```diff
@@ -96,13 +104,171 @@ func newState() *state {
 }
 
 // Marshal returns the json encoding of a terraform plan.
-func Marshal(s *states.State) ([]byte, error) {
+func Marshal(s *states.State, schemas *terraform.Schemas) ([]byte, error) {
 	if s.Empty() {
 		return nil, nil
 	}
 
 	output := newState()
 
+	// output.StateValues
+	err := output.marshalStateValues(s, schemas)
+	if err != nil {
+		return nil, err
+	}
+
 	ret, err := json.Marshal(output)
 	return ret, err
 }
+
+func (jsonstate *state) marshalStateValues(s *states.State, schemas *terraform.Schemas) error {
+	var sv stateValues
+	var err error
+
+	// only marshal the root module outputs
+	sv.Outputs, err = marshalOutputs(s.RootModule().OutputValues)
+	if err != nil {
+		return err
+	}
+
+	// use the state and module map to build up the module structure
+	sv.RootModule, err = marshalRootModule(s, schemas)
+	if err != nil {
+		return err
+	}
+
+	jsonstate.Values = sv
+	return nil
+}
+
+func marshalOutputs(outputs map[string]*states.OutputValue) (map[string]output, error) {
+	if outputs == nil {
+		return nil, nil
+	}
+
+	ret := make(map[string]output)
+	for k, v := range outputs {
+		ov, err := ctyjson.Marshal(v.Value, v.Value.Type())
+		if err != nil {
+			return ret, err
+		}
+		ret[k] = output{
```

Comment:
This struct doesn't have `json` tags on the fields, so it's coming through with initial-uppercase names in the JSON. Seems to also be true for `stateValues`.

Reply:
thank you! fixed!